restli-tools/src/main/java/com/linkedin/restli/tools/data/SchemaFormatTranslator.java

```diff
@@ -60,7 +60,7 @@
 {
   private static final Logger LOGGER = LoggerFactory.getLogger(SchemaFormatTranslator.class);
 
-  private static final Pattern LINE_END_SPACES = Pattern.compile("\\s+$", Pattern.MULTILINE);
+  private static final Pattern LINE_END_SPACES = Pattern.compile(" +$", Pattern.MULTILINE);
```

Comment:
I am sorry that my last suggestion introduced an error. I misunderstood "MULTILINE" mode. This mode actually match "$" to "any arbitrary" line's ending. (not current or each line)

Under MULTILINE mode, the correct regular expression should be `\s+?$` so that '\s' is not greedily matching consecutive empty line.
see example: https://regex101.com/r/TfmGhW/1

Reply:
Junchuan, np, and thanks for digging into the issue. 
In this case, I prefer to keep it simple and not change it. Since we are generating the PDL output, we control the whitespaces that needs trimming. And since we use only spaces, this is good.